src/libraries/Microsoft.Extensions.DependencyModel/tests/DependencyContextTests.cs

```diff
@@ -280,13 +281,20 @@ public void MergeMergesRuntimeGraph()
         [SkipOnTargetFramework(TargetFrameworkMonikers.NetFramework, "GetEntryAssembly() returns null")]
         public void DefaultWorksCorrectly()
         {
-            // only need to assert the context contains non-null properties.
-
             var context = DependencyContext.Default;
-            Assert.NotNull(context);
-            Assert.NotNull(context.RuntimeGraph);
-            Assert.NotNull(context.RuntimeLibraries);
-            Assert.NotNull(context.Target);
+            if (PlatformDetection.IsNativeAot)
```

Comment:
Does this work for single file deployments in general? If not:

```suggestion
            if (PlatformDetection.IsSingleFile)
```

Reply:
Good call. Fixed.